code/turf/turf.dm

```diff
@@ -997,6 +1003,25 @@ var/global/in_replace_with = 0
 				W.UpdateIcon()
 	return wall
 
+///Returns a turf to its initial path, if it is a simulated floor or wall. Returns FALSE if requested turf was not one of these initially.
+/turf/proc/ReplaceWithInitial()
+	var/oldpath = src.path_old
+	if(ispath(oldpath,/turf/simulated/floor) || ispath(oldpath,/turf/simulated/wall))
```

Comment:
This check feels like it should be outside the proc, otherwise it doesn't really do what it says it does.

Reply:
Moved ReplaceWith outside check so it can initial-ify anything, and put another check on the tool (check in the proc is still relevant for post-placement adjustment).